### PR TITLE
Action for marking and deleting stale branches

### PR DESCRIPTION
**Pull request by justus-camp**
## Description

This adds a GitHub action for marking stale branches and deleting them after notifying the owner of the branch. It will open a maximum of 20 issues and will mark a branch as stale after 120 days of no use. It won't automatically delete any protected branches and waits 365 days of no commits so that gives the owners of the branch ample time to decide if they need it.

### CODE REVIEW

**Comment by taylorsw04** [comment]
What's the logic behind a year before deletion? Additionally, does this open issues against the owner of the branch? Is that the issues referred to in the max of 20? If so, why limit that? Additionally, does this close the issues automatically after deleting the branch?

**Comment by justus-camp** [comment]
The year was based on feedback from Craig in a teams thread arguing against quick automated clean-up. The issue won't auto-assign the owner of the branch, but it tags the last committer to the branch (most likely the owner) in a comment on the issue. It does close the issue after auto-deletion, but I don't think it would automatically close the issue if you went in and manually deleted the branch after the issue was opened (not 100% sure here, I can go test on my fork/read the code). The max issues defaults to 20 but I'm open to configuring it higher. I was weary of opening 100 issues immediately upon running the tool. 

As an overall comment - if we want to invest some time in development on the GitHub action itself we could update it to be more in-line with whatever process we come up with.

File: .github/workflows/stale-branches.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Stale Branches
```
Comment (tylerbutler) [inline .github/workflows/stale-branches.yml:1]: Channeling @Josmithr... please add a header comment with the intent of the action, where to get more info or something, etc. pasting your PR description might be appropriate.

File: .github/workflows/stale-branches.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+permissions:
+  issues: write
+  contents: write
+
+jobs:
+  stale_branches:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Stale Branches
+      uses: crs-k/stale-branches@2864505cecb7aeea50767b9d2631ed550eb069b6
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply .github/workflows/stale-branches.yml:13]: I suggest ratchet because it makes maintenance a little easier.

**Review by taylorsw04 — APPROVED** [review]

**Pull request merged**